packages/editor-ui/src/views/WorkflowsView.vue

```diff
@@ -26,303 +56,250 @@ interface Filters {
+		option,
+	});
+	if (option === 'templates' && isSalesUser.value) {
+		trackCategoryLinkClick('Sales');
+	}
+};
+
+const trackCategoryLinkClick = (category: string) => {
+	telemetry.track(`User clicked Browse ${category} Templates`, {
+		role: usersStore.currentUserCloudInfo?.role,
+		active_workflow_count: workflowsStore.activeWorkflows.length,
+	});
+};
+
+const initialize = async () => {
+	loading.value = true;
+	await Promise.all([
+		usersStore.fetchUsers(),
+		workflowsStore.fetchAllWorkflows(route.params?.projectId as string | undefined),
+		workflowsStore.fetchActiveWorkflows(),
+	]);
+	loading.value = false;
+};
+
+const onClickTag = (tagId: string) => {
+	if (!filters.value.tags.includes(tagId)) {
+		filters.value.tags.push(tagId);
+	}
+};
+
+const saveFiltersOnQueryString = () => {
+	const query: { [key: string]: string } = {};
+
+	if (filters.value.search) {
+		query.search = filters.value.search;
+	}
+
+	if (typeof filters.value.status !== 'string') {
+		query.status = filters.value.status.toString();
+	}
+
+	if (filters.value.tags.length) {
+		query.tags = filters.value.tags.join(',');
+	}
+
+	if (filters.value.homeProject) {
+		query.homeProject = filters.value.homeProject;
+	}
+
+	void router.replace({
+		query: Object.keys(query).length ? query : undefined,
+	});
+};
+
+function isValidProjectId(projectId: string) {
+	return projectsStore.availableProjects.some((project) => project.id === projectId);
+}
+
+const setFiltersFromQueryString = async () => {
+	const { tags, status, search, homeProject } = route.query ?? {};
+
+	const filtersToApply: { [key: string]: string | string[] | boolean } = {};
+
+	if (homeProject && typeof homeProject === 'string') {
+		await projectsStore.getAvailableProjects();
+		if (isValidProjectId(homeProject)) {
+			filtersToApply.homeProject = homeProject;
+		}
+	}
+
+	if (search && typeof search === 'string') {
+		filtersToApply.search = search;
+	}
+
+	if (tags && typeof tags === 'string') {
+		await tagsStore.fetchAll();
+		const currentTags = tagsStore.allTags.map((tag) => tag.id);
+		const savedTags = tags.split(',').filter((tag) => currentTags.includes(tag));
+		if (savedTags.length) {
+			filtersToApply.tags = savedTags;
+		}
+	}
+
+	if (
+		status &&
+		typeof status === 'string' &&
+		[StatusFilter.ACTIVE.toString(), StatusFilter.DEACTIVATED.toString()].includes(status)
+	) {
+		filtersToApply.status = status === 'true';
+	}
+
+	if (Object.keys(filtersToApply).length) {
+		Object.assign(filters.value, filtersToApply);
+	}
+
+	void router.replace({ query: pickBy(route.query) });
+};
+
+sourceControlStore.$onAction(({ name, after }) => {
+	if (name !== 'pullWorkfolder') return;
+	after(async () => await initialize());
 });
 
-export default WorkflowsView;
+onMounted(async () => {
+	documentTitle.set(i18n.baseText('workflows.heading'));
+	await setFiltersFromQueryString();
+	void usersStore.showPersonalizationSurvey();
+});
 </script>
 
 <template>
```

Comment:
question(non-blocking): Looks like we are not using that `ref="layout"` in the ResourcesListLayout. Should we remove?